### PR TITLE
Failing to run prettier should block CI

### PR DESCRIPTION
**Pull request by acdlite**
https://github.com/facebook/react/pull/9101#issuecomment-286610833

`scripts/prettier/index.js write` will run prettier on source files. Run using `yarn prettier`.

`scripts/prettier/index.js` will throw if any source files are not formatted with prettier. We'll use this to block CI.

Based on similar script in Jest repo.

### CODE REVIEW

File: src/renderers/shared/fiber/ReactFiberUpdateQueue.js [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 'use strict';
 
-import type {Fiber} from 'ReactFiber';
+import type { Fiber } from 'ReactFiber';
```
Comment (spicyj) [inline src/renderers/shared/fiber/ReactFiberUpdateQueue.js]: Hey @acdlite – this project uses prettier to format all JavaScript code. Mind running `yarn prettier` and amending the changes to this PR?

File: scripts/prettier/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  const globPattern = patterns.length > 1
+    ? `{${patterns.join(',')}}`
+    : `${patterns.join(',')}`;
+  const files = glob.sync(globPattern, {ignore});
+
+  const args = Object.keys(defaultOptions).map(
+    k => `--${k}=${(options && options[k]) || defaultOptions[k]}`
+  );
+  args.push(`--${shouldWrite ? 'write' : 'l'} {${files.join(' ')}}`);
```
Comment (spicyj) [inline scripts/prettier/index.js]: This should not use shell escaping; can you change it to one of the methods that lets you pass args as an array?

File: src/renderers/shared/fiber/ReactFiberUpdateQueue.js (hunk 1 above)
Comment (acdlite) [inline src/renderers/shared/fiber/ReactFiberUpdateQueue.js]: Is this an automatic comment? :D

**Comment by gaearon** [comment]
Can we also add this to contributing instruction block in PR template and in Contributing docs section?

**Comment by acdlite** [comment]
@gaearon Good call Dan, added.

File: scripts/prettier/index.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    patterns: ['src/**/*.js'],
+    ignore: [
+      '**/third_party/**',
+      '**/node_modules/**',
+    ],
+  },
+};
+
+function exec(command, args) {
+  console.log('> ' + [command].concat(args).join(' '));
```
Comment (spicyj) [inline scripts/prettier/index.js]: I find this barely more helpful without the .join but if you like this then we can keep it.

File: docs/community/conferences.md [hunk 4]
```diff
@@ -34,6 +34,10 @@ July 10-11 in Portland, Oregon USA
 
 [Website](https://infinite.red/ChainReactConf) - [Twitter](https://twitter.com/chainreactconf)
 
+### React Rally
```
Comment (spicyj) [inline docs/community/conferences.md]: was this meant to be here?

File: scripts/prettier/index.js (hunk 3 above)
Comment (acdlite) [inline scripts/prettier/index.js]: Me too except it truncates some of the file names without it.

File: docs/community/conferences.md (hunk 4 above)
Comment (acdlite) [inline docs/community/conferences.md]: Rebase mistake I think

**Pull request merged**